In [2]:
import pandas
data = pandas.read_csv('thanksgiving.csv', encoding = 'Latin-1')
data

RespondentID Do you celebrate Thanksgiving?  \
0       4337954960                            Yes   
1       4337951949                            Yes   
2       4337935621                            Yes   
3       4337933040                            Yes   
4       4337931983                            Yes   
5       4337929779                            Yes   
6       4337924420                            Yes   
7       4337916002                            Yes   
8       4337914977                            Yes   
9       4337899817                            Yes   
10      4337899680                             No   
11      4337893416                            Yes   
12      4337888291                            Yes   
13      4337878450                            Yes   
14      4337878351                            Yes   
15      4337857295                            Yes   
16      4337856362                            Yes   
17      4337854106                            Yes   
18      4337844879                            Yes   
19      4337823612                            Yes   
20      4337820281                            Yes   
21      4337813502                            Yes   
22      4337809020                             No   
23      4337793158                            Yes   
24      4337792130                            Yes   
25      4337790002                            Yes   
26      4337783794                            Yes   
27      4337779071                            Yes   
28      4337778119                            Yes   
29      4337774090                            Yes   
...            ...                            ...   
1028    4335956575                             No   
1029    4335955478                            Yes   
1030    4335955206                            Yes   
1031    4335955152                            Yes   
1032    4335955050                             No   
1033    4335954681                            Yes   
1034    4335954394                            Yes   
1035    4335954376                            Yes   
1036    4335954286                             No   
1037    4335954207                            Yes   
1038    4335954131                            Yes   
1039    4335953888                            Yes   
1040    4335952833                            Yes   
1041    4335951505                            Yes   
1042    4335951437                            Yes   
1043    4335951082                            Yes   
1044    4335950917                            Yes   
1045    4335950654                            Yes   
1046    4335949486                            Yes   
1047    4335949169                            Yes   
1048    4335949112                            Yes   
1049    4335947496                            Yes   
1050    4335945415                            Yes   
1051    4335944854                            Yes   
1052    4335944115                             No   
1053    4335944082                            Yes   
1054    4335943173                            Yes   
1055    4335943060                            Yes   
1056    4335934708                            Yes   
1057    4335894916                            Yes   

     What is typically the main dish at your Thanksgiving dinner?  \
0                                                Turkey             
1                                                Turkey             
2                                                Turkey             
3                                                Turkey             
4                                              Tofurkey             
5                                                Turkey             
6                                                Turkey             
7                                                Turkey             
8                                                Turkey             
9                            

In [3]:
data.columns

Index(['RespondentID', 'Do you celebrate Thanksgiving?',
       'What is typically the main dish at your Thanksgiving dinner?',
       'What is typically the main dish at your Thanksgiving dinner? - Other (please specify)',
       'How is the main dish typically cooked?',
       'How is the main dish typically cooked? - Other (please specify)',
       'What kind of stuffing/dressing do you typically have?',
       'What kind of stuffing/dressing do you typically have? - Other (please specify)',
       'What type of cranberry saucedo you typically have?',
       'What type of cranberry saucedo you typically have? - Other (please specify)',
       'Do you typically have gravy?',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Brussel sprouts',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots',
       'Which of these side dishes aretypically served

In [4]:
data['Do you celebrate Thanksgiving?'].value_counts()

Yes    980
No      78
Name: Do you celebrate Thanksgiving?, dtype: int64

In [5]:
data = data[data['Do you celebrate Thanksgiving?']=='Yes']


In [6]:
data['What is typically the main dish at your Thanksgiving dinner?'].value_counts()

Turkey                    859
Other (please specify)     35
Ham/Pork                   29
Tofurkey                   20
Chicken                    12
Roast beef                 11
I don't know                5
Turducken                   3
Name: What is typically the main dish at your Thanksgiving dinner?, dtype: int64

In [7]:
data[data['What is typically the main dish at your Thanksgiving dinner?'] == 'Tofurkey']['Do you typically have gravy?']

4      Yes
33     Yes
69      No
72      No
77     Yes
145    Yes
175    Yes
218     No
243    Yes
275     No
393    Yes
399    Yes
571    Yes
594    Yes
628     No
774     No
820     No
837    Yes
860     No
953    Yes
Name: Do you typically have gravy?, dtype: object

In [8]:
apple_isnull = pandas.isnull(data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple'])
pumpkin_isnull = pandas.isnull(data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin'])
pecan_isnull = pandas.isnull(data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan'])
ate_pies = apple_isnull & pumpkin_isnull & pecan_isnull
ate_pies.value_counts()

False    876
True     104
dtype: int64

In [9]:
def converter(str_value):
    if pandas.isnull(str_value) is True:
        return None
    else:
        str_value = str_value.split(' ')[0]
        str_value = str_value.replace('+', '')
        
        return int(str_value)
data['int_age'] = data['Age'].apply(converter)
data['int_age'].describe()
    
    

/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  if __name__ == '__main__':


count    947.000000
mean      40.089757
std       15.352014
min       18.000000
25%       30.000000
50%       45.000000
75%       60.000000
max       60.000000
Name: int_age, dtype: float64

The age range of participants are fairly even distributed. The lower bound of age is taken to consideration only. 

In [10]:
data['How much total combined money did all members of your HOUSEHOLD earn last year?'].value_counts()

$25,000 to $49,999      166
$75,000 to $99,999      127
$50,000 to $74,999      127
Prefer not to answer    118
$100,000 to $124,999    109
$200,000 and up          76
$10,000 to $24,999       60
$0 to $9,999             52
$125,000 to $149,999     48
$150,000 to $174,999     38
$175,000 to $199,999     26
Name: How much total combined money did all members of your HOUSEHOLD earn last year?, dtype: int64

In [11]:
def a_integer(str_value):
    if pandas.isnull(str_value) is True:
        return None 
    str_value = str_value.split(' ')[0]
    if str_value == 'Prefer':
        return None
    str_value = str_value.replace('$', '')
    str_value = str_value.replace(',', '')
    return int(str_value)

data['int_income'] = data['How much total combined money did all members of your HOUSEHOLD earn last year?'].apply(a_integer)
data['int_income'].describe()

/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  # This is added back by InteractiveShellApp.init_path()


count       829.000000
mean      75965.018094
std       59068.636748
min           0.000000
25%       25000.000000
50%       75000.000000
75%      100000.000000
max      200000.000000
Name: int_income, dtype: float64

Due to high standard deviation, we can say there is a sharp difference between total income of participants. However the average income is quite high, $75,965. 

In [12]:
travel_u150k = data['int_income'] < 150000
data[travel_u150k]['How far will you travel for Thanksgiving?'].value_counts(normalize = True)

Thanksgiving is happening at my home--I won't travel at all                         0.407837
Thanksgiving is local--it will take place in the town I live in                     0.294630
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    0.217707
Thanksgiving is out of town and far away--I have to drive several hours or fly      0.079826
Name: How far will you travel for Thanksgiving?, dtype: float64

In [13]:
travel_u150k = data['int_income'] > 150000
data[travel_u150k]['How far will you travel for Thanksgiving?'].value_counts(normalize = True)

Thanksgiving is happening at my home--I won't travel at all                         0.480392
Thanksgiving is local--it will take place in the town I live in                     0.245098
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    0.156863
Thanksgiving is out of town and far away--I have to drive several hours or fly      0.117647
Name: How far will you travel for Thanksgiving?, dtype: float64

Percentage of people with lower income who have to drive for a few hour or less to celebrate Thanksgiving is higher. On the other hand, there is a decent amount of wealthier people who won't travel at all for Thanksgiving. 
Explanation for the first comparision could be students and young adults who do not have high income, work/live in a different city and tend to comeback to there hometown. This is different for the high-income group. They are most likely to settle down at their current place and celebrate Thanksgiving without traveling much. 

In [14]:
data.pivot_table(index = 'Have you ever tried to meet up with hometown friends on Thanksgiving night?', columns = 'Have you ever attended a "Friendsgiving?"', values = 'int_age')

Have you ever attended a "Friendsgiving?"                  No        Yes
Have you ever tried to meet up with hometown fr...                      
No                                                  42.283702  37.010526
Yes                                                 41.475410  33.976744

Younger people are more willing to attend Friendsgiving or meet up with their hometown friends on that day. 

In [23]:
#Most common dessert (besides pie)
desserts = data.filter(regex = '.*desserts.*')
pandas.value_counts(desserts.values.flatten())

None                                                      295
Ice cream                                                 266
Cookies                                                   204
Cheesecake                                                191
Other (please specify)                                    134
Brownies                                                  128
Apple cobbler                                             110
Peach cobbler                                             103
Carrot cake                                                72
Fudge                                                      43
Blondies                                                   16
pie                                                        13
Pie                                                        12
pies                                                        6
pumpkin pie                                                 4
Pumpkin pie                                                 4
chocolat

The most popular dessert besides pie during Thanksgiving is ice cream. 

In [16]:
#Most common side dishes
side_dish= data.filter(regex = '.*side dishes.*')
pandas.value_counts(side_dish.values.flatten())

Mashed potatoes                                                                   817
Rolls/biscuits                                                                    766
Green beans/green bean casserole                                                  686
Yams/sweet potato casserole                                                       631
Corn                                                                              464
Carrots                                                                           242
Cornbread                                                                         235
Fruit salad                                                                       215
Vegetable salad                                                                   209
Macaroni and cheese                                                               206
Squash                                                                            171
Brussel sprouts                                       

In [17]:
#Most common pie
pies= data.filter(regex = '.*pie.*')
pandas.value_counts(pies.values.flatten())

Pumpkin                                           729
Apple                                             514
Pecan                                             342
Sweet Potato                                      152
Chocolate                                         133
Cherry                                            114
Other (please specify)                             71
None                                               40
Key lime                                           39
Coconut cream                                      36
Buttermilk                                         35
Peach                                              34
Chess                                               3
mincemeat                                           2
minced meat                                         2
Lemon                                               2
peanut butter                                       2
mince meat                                          2
Mincemeat                   

A common full Thanksgiving meal includes of Turkey, Mashed potatoes, Biscuits, Green bean or Yam casserole and Pumpkin pie. Most of them don't have desserts. 

In [18]:
#People work on Thanksgiving
work_on_thanksgiving = data['Will you employer make you work on Black Friday?']
pandas.value_counts(work_on_thanksgiving)


Yes              43
No               20
Doesn't apply     7
Name: Will you employer make you work on Black Friday?, dtype: int64

In [19]:
#Regional pattern in dinner menu
regional_main = data[['What is typically the main dish at your Thanksgiving dinner?','What is typically the main dish at your Thanksgiving dinner? - Other (please specify)']].groupby(data['US Region'])
regional_main.describe()

What is typically the main dish at your Thanksgiving dinner?  \
                                                                          count   
US Region                                                                         
East North Central                                                145             
East South Central                                                 56             
Middle Atlantic                                                   145             
Mountain                                                           41             
New England                                                        55             
Pacific                                                           130             
South Atlantic                                                    203             
West North Central                                                 71             
West South Central                                                 85             

                                        \
                   unique     top freq   
US Region                                
East North Central      4  Turkey  135   
East South Central      4  Turkey   50   
Middle Atlantic         7  Turkey  130   
Mountain                4  Turkey   37   
New England             4  Turkey   51   
Pacific                 7  Turkey  107   
South Atlantic          6  Turkey  181   
West North Central      6  Turkey   60   
West South Central      5  Turkey   77   

                   What is typically the main dish at your Thanksgiving dinner? - Other (please specify)  \
                                                                                                   count   
US Region                                                                                                  
East North Central                                                  5                                      
East South Central                                                  4                                      
Middle Atlantic                                                     4                                      
Mountain                                                            0                                      
New England                                                         1                                      
Pacific                                                             9                                      
South Atlantic                                                      6                                      
West North Central                                                  3                                      
West South Central                                                  3                                      

                                                                              \
                   unique                                                top   
US Region                                                                      
East North Central      5                                        head cheese   
East South Central      4  This year we are having baked lasagna, diff ev...   
Middle Atlantic         4                              Homemade vegan entree   
Mountain                0                                                NaN   
New England             1                                   always different   
Pacific                 9                                             Varies   
South Atlantic          6                                            seafood   
West North Central      3  Turkey & Ham or Seafood Hotdish (a secret fami...   
West South Central      3                                          Prime Rib   

                         
                   freq  
US Region                
East North Central    1  
East South Central    1  
Middle Atlantic       1  
Mountain            NaN  
New England           1  
Pacific               1  
South Atlantic        1  

In [20]:
regional_side = side_dish.groupby(data['US Region'])
pandas.set_option('display.max_columns', None)
regional_side.describe()

Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Brussel sprouts  \
                                                                                                                                         count   
US Region                                                                                                                                        
East North Central                                                 22                                                                            
East South Central                                                  7                                                                            
Middle Atlantic                                                    41                                                                            
Mountain                                                            6                                                                            
New England                                                        12                                                                            
Pacific                                                            29                                                                            
South Atlantic                                                     27                                                                            
West North Central                                                  3                                                                            
West South Central                                                  4                                                                            

                                                 \
                   unique              top freq   
US Region                                         
East North Central      1  Brussel sprouts   22   
East South Central      1  Brussel sprouts    7   
Middle Atlantic         1  Brussel sprouts   41   
Mountain                1  Brussel sprouts    6   
New England             1  Brussel sprouts   12   
Pacific                 1  Brussel sprouts   29   
South Atlantic          1  Brussel sprouts   27   
West North Central      1  Brussel sprouts    3   
West South Central      1  Brussel sprouts    4   

                   Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots  \
                                                                                                                                 count   
US Region                                                                                                                                
East North Central                                                 28                                                                    
East South Central                                                 15                                                                    
Middle Atlantic                                                    45                                                                    
Mountain                                                           11                                                                    
New England                                                        25                                                                    
Pacific                                                            30                                                                    
South Atlantic                                                     48                                                                    
West North Central                                                 12                                                                    
West South Central                                                 21                                                                  

The most common side dish in most of the regions is Mashed potatoes. However, in Southern regions such as East South Central, West South Central and South Atlantic, Green beans/green bean casserole, Macaroni and cheese or rolls/biscuits are also very popular. 

In [22]:
regional_pies = pies.groupby(data['US Region'])
pandas.set_option('display.max_columns', None)
regional_pies.describe()

Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple  \
                                                                                                                      count   
US Region                                                                                                                     
East North Central                                                 75                                                         
East South Central                                                 19                                                         
Middle Atlantic                                                   106                                                         
Mountain                                                           19                                                         
New England                                                        41                                                         
Pacific                                                            74                                                         
South Atlantic                                                    101                                                         
West North Central                                                 37                                                         
West South Central                                                 32                                                         

                                       \
                   unique    top freq   
US Region                               
East North Central      1  Apple   75   
East South Central      1  Apple   19   
Middle Atlantic         1  Apple  106   
Mountain                1  Apple   19   
New England             1  Apple   41   
Pacific                 1  Apple   74   
South Atlantic          1  Apple  101   
West North Central      1  Apple   37   
West South Central      1  Apple   32   

                   Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Buttermilk  \
                                                                                                                           count   
US Region                                                                                                                          
East North Central                                                  3                                                              
East South Central                                                  6                                                              
Middle Atlantic                                                     2                                                              
Mountain                                                            0                                                              
New England                                                         1                                                              
Pacific                                                             2                                                              
South Atlantic                                                      4                                                              
West North Central                                                  2                                                              
West South Central                                                 13                                                              

                                            \
                   unique         top freq   
US Region                                    
East North Central      1  Buttermilk    3   
East South Central      1  Buttermilk    6   
Middle Atlantic         1  Buttermilk    2   
Mountain                0         NaN  NaN   
New England             1  Buttermilk    1   
Pacific                 1  Buttermilk    2   
So

Southern regions have more diversed choices of pies while pumpkin remains the most common pie during Thanksgiving.

In [24]:
#Most common dessert besides pie
regional_desserts = desserts.groupby(data['US Region'])
pandas.set_option('display.max_columns', None)
regional_desserts.describe()

Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Apple cobbler  \
                                                                                                                                   count   
US Region                                                                                                                                  
East North Central                                                 13                                                                      
East South Central                                                  9                                                                      
Middle Atlantic                                                    23                                                                      
Mountain                                                            1                                                                      
New England                                                         2                                                                      
Pacific                                                            18                                                                      
South Atlantic                                                     24                                                                      
West North Central                                                  7                                                                      
West South Central                                                 11                                                                      

                                               \
                   unique            top freq   
US Region                                       
East North Central      1  Apple cobbler   13   
East South Central      1  Apple cobbler    9   
Middle Atlantic         1  Apple cobbler   23   
Mountain                1  Apple cobbler    1   
New England             1  Apple cobbler    2   
Pacific                 1  Apple cobbler   18   
South Atlantic          1  Apple cobbler   24   
West North Central      1  Apple cobbler    7   
West South Central      1  Apple cobbler   11   

                   Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Blondies  \
                                                                                                                              count   
US Region                                                                                                                             
East North Central                                                  1                                                                 
East South Central                                                  2                                                                 
Middle Atlantic                                                     0                                                                 
Mountain                                                            0                                                                 
New England                                                         6                                                                 
Pacific                                                             3                                                                 
South Atlantic                                                      3                                                                 
West North Central                                                  0                                                                 
West South Central                                                  1                                                                 

                                          \
                   unique       top freq   
US Region                               

Participants from Middle Atlantic and New England have Ice cream as desserts.